projects/packages/connection/src/class-rest-connector.php

```diff
@@ -791,7 +790,10 @@ public static function jetpack_register_permission_check() {
 	 * @return \WP_REST_Response|WP_Error
 	 */
 	public function connection_register( $request ) {
-		if ( ! wp_verify_nonce( $request->get_param( 'registration_nonce' ), 'jetpack-registration-nonce' ) ) {
+		$is_app = did_action( 'application_password_did_authenticate' );
```

Comment:
Do you think it might be worth making this a bit more generic? Instead of checking whether the current user is authenticated via a application password to skip the nonce verification, check if `did_action( 'auth_cookie_valid` )` aka cookie authentication is used and only verify the nonce in this case.